### PR TITLE
Integration of relational database.

### PR DESCRIPTION
**Pull request by FKGSOFTWARE**
Implemented a relational database to take the place of the in-memory solutions as well as provide a more robust and scalable foundation. 

Functionality spread from the agent.py file to other files within the LTM folder. 

All semantic memory functionality is maintained and accessed through the hybrid memory. 

datasets -> local_datasets.

requirements updated for use with SQLAlchemy.

### CODE REVIEW

File: dataset_interfaces/factory.py [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-from datasets.name import NamesDataset
-from datasets.name_list import NameListDataset
-from datasets.sally_ann import SallyAnneDataset
-from datasets.shopping import ShoppingDataset
-from datasets.spy_meeting import SpyMeetingDataset
-from datasets.trigger_response import TriggerResponseDataset
-from datasets.kv import KVPairsDataset
-from datasets.chapterbreak import ChapterBreakDataset
-from datasets.restaurant import RestaurantDataset
+from local_datasets.conficting_personal_information import ConflictingPersonalInformationDataset
```
Comment (dcasbol) [inline dataset_interfaces/factory.py:4]: if reasonable, it would be good to not change this, and so also keep the num. of changes low.

Comment (FKGSOFTWARE) [reply dataset_interfaces/factory.py:4]: Right oh I will revert this. That will remove a lot of the changes.

File: ltm/archived/baseline-scratchpad-29_7.py [hunk 2]
```diff
@@ -1,462 +0,0 @@
...
-import json
-import logging
-import uuid
-import ltm.scratchpad as sp
-from dataclasses import dataclass, field
-from copy import deepcopy
-from json import JSONDecodeError
-from typing import List, Callable, Optional, Any, Union
-from goodai.helpers.json_helper import sanitize_and_parse_json, SimpleJSONEncoder, SimpleJSONDecoder
-from goodai.ltm.agent import LTMAgentConfig, Message
```
Comment (JosephDavidsonKSWH) [inline ltm/archived/baseline-scratchpad-29_7.py:11]: Why is this archived file deleted?

File: requirements.txt
Comment (JosephDavidsonKSWH) [inline requirements.txt:1]: Why is this a binary file?

File: ltm/scratchpad.py
Comment (JosephDavidsonKSWH) [inline ltm/scratchpad.py:1]: Do we need to erase all the previous ltm tools? Its not really affecting anything to keep it in

File: ltm/archived/baseline-scratchpad-29_7.py (hunk 2 above)
Comment (FKGSOFTWARE) [reply ltm/archived/baseline-scratchpad-29_7.py:11]: archived file reinstated. Moved current dev version of scratchpad and agent to archived/

File: requirements.txt
Comment (FKGSOFTWARE) [reply requirements.txt:1]: Deleted and remade, hopefully addressed the issue. Still a .txt locally.

File: ltm/scratchpad.py
Comment (FKGSOFTWARE) [reply ltm/scratchpad.py:1]: Moved to ltm/archived/

File: requirements.txt
Comment (JosephDavidsonKSWH) [inline requirements.txt:1]: Errr... I am not a fan of this venv dump. @dcasbol What do you think

Comment (dcasbol) [reply requirements.txt:1]: oh yeah... I'd keep it to the bare minimum. Most of the entries look like they're dependencies of other packages. @FKGSOFTWARE could you put only the key requirements? dependencies should be taken care of automatically by the pip installer.

Comment (FKGSOFTWARE) [reply requirements.txt:1]: Right oh, I'll sort this out

File: requirements.txt [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
-
+click
+anthropic
+pymemgpt==0.3.16
+openai
+faker~=20.1.0
+colorama~=0.4.6
+pytest
+time-machine==2.13.0
+tiktoken==0.5.2
```
Comment (JosephDavidsonKSWH) [inline requirements.txt:9]: This should be higher

File: requirements.txt [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
+goodai-ltm==0.4.4
+gdown==4.7.1  # For downloading files from google drive (some datasets use it)
+tqdm  # Showing progress bar during some downloads
+pyyaml~=6.0.1
+jinja2~=3.1.2
+Werkzeug==2.2.2  # It's crucial to specify this package's version to make flask work
+flask==2.2.3
+browser-cookie3
+rouge_score==0.1.2
+litellm==1.40.8  # Unifies interface from most LLM APIs
```
Comment (JosephDavidsonKSWH) [inline requirements.txt:19]: And so should this

File: requirements.txt [hunk 5]
```diff
@@ -1,25 +1,25 @@
...
-litellm>=1.41.13  # Unifies interface from most LLM APIs
-zstd==1.5.5.1  # To decompress chapterbreak
-humanize==4.9.0  # Shows data in human-readable form e.g. timedelta as "1 hour and 23 seconds"
-transformers==4.40.2  # Required to load some HuggingFace tokenizers
-matplotlib==3.8.4  # To run plotting scripts under dev_scripts
-google-generativeai==0.4.1  # Gemini API
-
+click
+anthropic
+pymemgpt==0.3.16
```
Comment (JosephDavidsonKSWH) [inline requirements.txt:3]: delete

**Review by dcasbol — APPROVED** [review]

**Pull request closed without merging**